openml_OS/models/api/v1/Api_data.php

```diff
@@ -547,18 +580,14 @@ private function status_update($data_id, $status) {
   }
 
   private function data_features($data_id) {
-    if($data_id == false) {
-      $this->returnError(270, $this->version);
-      return;
-    }
     $dataset = $this->Dataset->getById($data_id);
-    if( $dataset === false ) {
+    if ($dataset === false) {
       $this->returnError(271, $this->version);
       return;
     }
 
-    if($dataset->visibility != 'public' && $dataset->uploader != $this->user_id) {
-      $this->returnError(271, $this->version); // Add special error code for this case?
+    if ($dataset->visibility != 'public' && !($dataset->uploader == $this->user_id || $this->user_has_admin_rights)) {
```

Comment:
I don't understand this. What does visibility have to do with the possibility to remove a dataset? Is it not allowed to remove a non-public dataset?

Reply:
this ensures that only:
- features of public datasets can be seen
- admins can see non-public datasets
- users can see non-public datasets (when they own)

that's what i intended to write down. please verify if that happens.

Reply:
Ah sorry, looked at the wrong function name in the diff. Looks good.